zds/searchv2/views.py

```diff
@@ -82,55 +100,54 @@ def __init__(self, **kwargs):
         results = []
-        if self.index_manager.connected_to_es and self.search_query:
+        if self.index_manager.connected_to_search and self.search_query:
             self.authorized_forums = get_authorized_forums(self.request.user)
 
-            search_queryset = Search()
+            search_parameters = {
+                "q": self.search_query,
+                "query_by": "title,description",
+            }
+
             if len(excluded_content_ids) > 0 and excluded_content_ids != [""]:
-                search_queryset = search_queryset.exclude("terms", content_pk=excluded_content_ids)
-            query = Match(_type="publishedcontent") & MultiMatch(
-                query=self.search_query, fields=["title", "description"]
-            )
-
-            functions_score = [
-                {
-                    "filter": Match(content_type="TUTORIAL"),
-                    "weight": settings.ZDS_APP["search"]["boosts"]["publishedcontent"]["if_tutorial"],
-                },
-                {
-                    "filter": Match(content_type="ARTICLE"),
-                    "weight": settings.ZDS_APP["search"]["boosts"]["publishedcontent"]["if_article"],
-                },
-                {
-                    "filter": Match(content_type="OPINION"),
-                    "weight": settings.ZDS_APP["search"]["boosts"]["publishedcontent"]["if_opinion"],
-                },
-            ]
+                search_parameters["filter_by"] = self._add_a_negative_numerical_filter(
+                    "content_pk", excluded_content_ids
+                )
 
-            scored_query = FunctionScore(query=query, boost_mode="multiply", functions=functions_score)
-            search_queryset = search_queryset.query(scored_query)[:10]
+            hits = client.collections["publishedcontent"].documents.search(search_parameters)["hits"][:10]
 
             # Build the result
-            for hit in search_queryset.execute():
+            for hit in hits:
+                document = hit["document"]["_source"]
                 result = {
-                    "id": hit.content_pk,
-                    "pubdate": hit.publication_date,
-                    "title": str(hit.title),
-                    "description": str(hit.description),
+                    "id": document["content_pk"],
+                    "pubdate": document["publication_date"],
+                    "title": str(document["title"]),
+                    "description": str(document["description"]),
                 }
                 results.append(result)
-
         data = {"results": results}
 
         return HttpResponse(json_handler.dumps(data), content_type="application/json")
 
+    def _add_a_negative_numerical_filter(self, field, values):
+        """
+        Add a filter to the current filter, this filter is used for numerical negation
+        Indeed, in 0.24.0, Typesense doesn't allow numerical negation
```

Comment:
```suggestion
        Indeed, in its version 0.24.0, Typesense doesn't allow numerical negation
```

Ce n'est pas clair quel est le problème et qu'est-ce que vous voulez vraiment faire. En lisant sans trop réfléchir, j'avais compris que c'était pour filtrer suivant des nombres négatifs. En fait c'est pour exclure un nombre ? Utiliser le terme `exclude` serait sans doute alors plus parlant que `negation` (*exclude  a numerical value*).

Reply:
On voulait bien parler de l'exclusion d'un nombre. Je vais faire la modification.